core/model/objects/domain.hpp

```diff
@@ -19,19 +19,22 @@ limitations under the License.
 #define CORE_DOMAIN_OBJECTS_DOMAIN_HPP_
 
 #include <string>
-#include "object.hpp"
+#include <memory>
 #include "../../service/json_parse.hpp"
 
-namespace domain {
+namespace object {
 
-class Domain : public AbsObject{
+class Domain{
     std::string ownerPublicKey;
     std::string name;
-
 public:
     Domain(
-        std::string ownerPublicKey,
-        std::string name
+        const std::string& ownerPublicKey,
+        const std::string& name
+    );
+
+    Domain(
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Class 'Domain' has a constructor with 1 argument that is not explicit.](https://www.codacy.com/app/soramitsu/iroha/file/4478128053/issues/source?bid=3712491&fileBranchId=3923596#l36)